pkg/grafana/library-elements-handler.go

```diff
@@ -0,0 +1,225 @@
+	path := filepath.Join(dir, libraryElementGlob)
+	return filepath.Glob(path)
+}
+
+// ResourceFilePath returns the location on disk where a resource should be updated
+func (h *LibraryElementHandler) ResourceFilePath(resource grizzly.Resource, filetype string) string {
+	kind := "element"
+	t := resource.GetSpecValue("kind").(float64)
+
+	switch t {
+	case 1:
+		kind = "panel"
+	case 2:
+		kind = "variable"
+	}
+	return fmt.Sprintf(libraryElementPattern, kind, resource.Name(), filetype)
+}
+
+// Parse parses a manifest object into a struct for this resource type
+func (h *LibraryElementHandler) Parse(m manifest.Manifest) (grizzly.Resources, error) {
+	resource := grizzly.Resource(m)
+	resource.SetSpecString("uid", resource.Name())
+	return grizzly.Resources{resource}, nil
+}
+
+// Unprepare removes unnecessary elements from a remote resource ready for presentation/comparison
+func (h *LibraryElementHandler) Unprepare(resource grizzly.Resource) *grizzly.Resource {
+	resource.DeleteSpecKey("meta")
+	resource.DeleteSpecKey("version")
+	resource.DeleteSpecKey("id")
+	return &resource
+}
+
+// Prepare gets a resource ready for dispatch to the remote endpoint
+func (h *LibraryElementHandler) Prepare(existing, resource grizzly.Resource) *grizzly.Resource {
+	if existing != nil {
+		val := existing.GetSpecValue("version")
+		resource.SetSpecValue("version", val)
+	}
+	resource.DeleteSpecKey("meta")
+	return &resource
+}
+
+// GetUID returns the UID for a resource
+func (h *LibraryElementHandler) GetUID(resource grizzly.Resource) (string, error) {
+	return resource.Name(), nil
+}
+
+// GetByUID retrieves JSON for a resource from an endpoint, by UID
+func (h *LibraryElementHandler) GetByUID(UID string) (*grizzly.Resource, error) {
+	resource, err := h.getRemoteLibraryElement(UID)
+	if err != nil {
+		return nil, fmt.Errorf("Error retrieving library element %s: %w", UID, err)
+	}
+
+	return resource, nil
+}
+
+// GetRemote retrieves an element as a resource
+func (h *LibraryElementHandler) GetRemote(resource grizzly.Resource) (*grizzly.Resource, error) {
+	return h.getRemoteLibraryElement(resource.Name())
+}
+
+// ListRemote retrieves as list of UIDs of all remote resources
+func (h *LibraryElementHandler) ListRemote() ([]string, error) {
+	return h.listElements()
+}
+
+// Add pushes a new element to Grafana via the API
+func (h *LibraryElementHandler) Add(resource grizzly.Resource) error {
+	return h.createElement(resource)
+}
+
+// Update pushes an element to Grafana via the API
+func (h *LibraryElementHandler) Update(existing, resource grizzly.Resource) error {
+	return h.updateElement(existing, resource)
+}
+
+func (h *LibraryElementHandler) listElements() ([]string, error) {
+	params := library.NewGetLibraryElementsParams()
+	client, err := h.Provider.(ClientProvider).Client()
+	if err != nil {
+		return nil, err
+	}
+	elemsOK, err := client.LibraryElements.GetLibraryElements(params, nil)
```

Comment:
You'll need to rebase following the OpenAPI client update. This function likely changed (no more params)

Reply:
done!